provider/schema/execution.go

```diff
@@ -150,55 +195,95 @@ func (e ExecutionData) resolveResources(ctx context.Context, meta ClientMeta, pa
 	// only top level tables should cascade, disable delete is turned on.
 	// if we didn't disable delete all data should be wiped before resolve)
 	shouldCascade := parent == nil && e.disableDelete
-	if err := e.Db.CopyFrom(ctx, resources, shouldCascade, e.extraFields); err != nil {
-		e.Logger.Warn("failed copy-from to db", "error", err)
-		// fallback insert, copy from sometimes does problems so we fall back with insert
-		if err := e.Db.Insert(ctx, e.Table, resources); err != nil {
-			e.Logger.Error("failed insert to db", "error", err)
-			return err
-		}
+	var err error
+	resources, err = e.copyDataIntoDB(ctx, resources, shouldCascade)
+	if err != nil {
+		return err
 	}
 
-	// Finally resolve relations of each resource
+	// Finally, resolve relations of each resource
 	for _, rel := range e.Table.Relations {
 		meta.Logger().Debug("resolving table relation", "table", e.Table.Name, "relation", rel.Name)
 		for _, r := range resources {
 			// ignore relation resource count
 			_, err := e.WithTable(rel).ResolveTable(ctx, meta, r)
 			if err != nil {
-				return err
+				if partialFetchErr := e.checkPartialFetchError(err, r, "resolve relation error"); partialFetchErr != nil {
+					return partialFetchErr
+				}
 			}
 		}
 	}
 	return nil
 }
 
-func (e ExecutionData) resolveResourceValues(ctx context.Context, meta ClientMeta, resource *Resource) (err error) {
+func (e *ExecutionData) copyDataIntoDB(ctx context.Context, resources Resources, shouldCascade bool) (Resources, error) {
+	partialFetchErrorOccurred := false
+	if err := e.Db.CopyFrom(ctx, resources, shouldCascade, e.extraFields); err != nil {
```

Comment:
invert this, so if copyfrom doesn't fail, return nil. if bulk insert doesn't fail return nil. if we are not partial fetch return err, insert one by one

Reply:
Thanks!